optimizer/plan/planbuilder_join.go

```diff
@@ -209,7 +209,7 @@ func (p *joinPath) attachCondition(condition ast.ExprNode, availablePaths []*joi
 		p.filterRate *= filterRate
 		return true
 	}
-	if p.inner.attachCondition(condition, append(availablePaths, p.outer)) {
+	if p.inner.attachCondition(condition, availablePaths) {
```

Comment:
It will never be a on condition. On condition will be processed in newOuterJoinPath.